auth/webhook/client.go

```diff
@@ -1,56 +1,49 @@
 package webhook
 
 import (
-	"net"
-
 	auth2 "github.com/containerssh/libcontainerssh/auth"
 	"github.com/containerssh/libcontainerssh/config"
 	"github.com/containerssh/libcontainerssh/internal/auth"
 	"github.com/containerssh/libcontainerssh/internal/geoip/dummy"
 	"github.com/containerssh/libcontainerssh/internal/metrics"
 	"github.com/containerssh/libcontainerssh/log"
+	"github.com/containerssh/libcontainerssh/metadata"
 )
 
 type Client interface {
 	// Password authenticates with a password from the client. It returns a bool if the authentication as successful
 	// or not. If an error happened while contacting the authentication server it will return an error.
 	Password(
-		username string,
+		metadata metadata.ConnectionAuthenticatingMetadata,
 		password []byte,
-		connectionID string,
-		remoteAddr net.IP,
-	) AuthenticationContext
+	) AuthenticationResponse
 
 	// PubKey authenticates with a public key from the client. It returns a bool if the authentication as successful
 	// or not. If an error happened while contacting the authentication server it will return an error.
 	PubKey(
-		username string,
-		pubKey string,
-		connectionID string,
-		remoteAddr net.IP,
-	) AuthenticationContext
+		metadata metadata.ConnectionAuthenticatingMetadata,
+		pubKey auth2.PublicKey,
+	) AuthenticationResponse
 }
 
-// AuthenticationContext holds the results of an authentication.
-type AuthenticationContext interface {
+// AuthenticationResponse holds the results of an authentication.
+type AuthenticationResponse interface {
 	// Success must return true or false of the authentication was successful / unsuccessful.
 	Success() bool
-	// Error returns the error that happened during the authentication.
+	// Error returns the error that happened during the authentication. This is useful for returning detailed error
+	// message.
 	Error() error
 	// Metadata returns a set of metadata entries that have been obtained during the authentication.
-	Metadata() *auth2.ConnectionMetadata
+	Metadata() metadata.ConnectionAuthenticatedMetadata
 }
 
 // NewTestClient creates a new copy of a client usable for testing purposes.
 func NewTestClient(cfg config.AuthWebhookClientConfig, logger log.Logger) (Client, error) {
-	clientConfig := config.AuthConfig{
-		Method:  config.AuthMethodWebhook,
-		Webhook: cfg,
-	}
 	metricsCollector := metrics.New(dummy.New())
 
-	authClient, err := auth.NewHttpAuthClient(
-		clientConfig,
+	authClient, err := auth.NewWebhookClient(
+		auth.AuthenticationTypeAll,
+		cfg,
```

Comment:
I have discovered a possible validation error in the associated structure.

The problem is in the authentication timeout validation, since `c.Timeout` has been renamed to `c.AuthTimeout`.

Ref link: https://github.com/ContainerSSH/libcontainerssh/blob/a0ceab26028f72e1bc717141c7bef988f881e962/config/auth.go#L409-L411